rust/hyperlane-base/src/types/local_storage.rs

```diff
@@ -68,6 +72,19 @@ impl CheckpointSyncer for LocalStorage {
         }
     }
 
+    async fn fetch_checkpoint_with_message_id(
+        &self,
+        index: u32,
+    ) -> Result<Option<SignedCheckpointWithMessageId>> {
+        match tokio::fs::read(self.checkpoint_with_message_id_file_path(index)).await {
+            Ok(data) => {
+                let checkpoint = serde_json::from_slice(&data)?;
+                Ok(Some(checkpoint))
+            }
+            _ => Ok(None),
+        }
```

Comment:
I realize you were just mimicking what was already there but this was bothering me so...

```suggestion
        let Ok(data) = tokio::fs::read(self.checkpoint_with_message_id_file_path(index)).await else {
            return Ok(None)
        };
        let checkpoint = serde_json::from_slice(&data)?;
        Ok(Some(checkpoint))
```